eng/build.ps1

```diff
@@ -166,13 +168,21 @@ function Process-Arguments() {
     $script:applyOptimizationData = $false
   }
 
+  if ($binaryLogName -ne "") {
+    $script:binaryLog = $true
```

Comment:
Initially I required specifying both options so `-binaryLog -binaryLogName "Build.binlog"` but that just looked too verbose to me. Decided it was concise to allow `-binaryLogName` to imply `-binaryLog`